Automattic/woocommerce-payments
# Update Transaction Details breakdown: rename labels and add fee breakdown tooltip when disputed

Fixes #6974

> **Warning**
> This PR is **NOT** behind a feature flag


#### Changes proposed in this Pull Request

This PR updates the Transaction Details screen with the following changes:
* Update transaction breakdown labels:
	* ~Refunded~ → Deducted
		* only for disputed transactions where a dispute fee has been applied and not reversed
	* ~Fee~ → Fees
	* Change style to uppercase
* A click tooltip that only shows if there is a Dispute fee in addition to the transaction fee. Tooltip content includes fee breakdown:
	* Transaction fee
	* Dispute fee
	* Total fee

**Note:** HorizontalList uppercase labels were updated in #7126 

**Before:** (`develop`)
<img width="1032" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/b565cf55-30be-4d12-a0ce-1e6c140137e2">


**After:**
<img width="1033" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/8d8e4e2f-bb64-48f4-adcd-7578b9acbb17">

Fees tooltip content, multi-currency enabled:
<img width="1056" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/dad874aa-ceeb-4fd8-b1b4-7b4320f37b3c">


**Design:**
<img width="1001" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/2644bd8d-78a2-4fc6-91e6-ccd79f5bf4f3">


Additionally, some other code improvements:
* Add the `reporting_category` property to the `BalanceTransaction` type.
* Fix a `ClickTooltip` bug preventing the `className` prop from being applied to the tooltip content element.

#### Testing instructions





* Create a disputed transaction using the card `4000000000000259`.
* Navigate to this disputed via the Payments → Transactions list screen.
* Ensure the Fees tooltip renders, along with the correct fee breakdown.
	* Ensure the **Transaction fee** matches the **Fee** total found on the timeline further down this transaction details screen:
		<img width="400" alt="image" src="https://github.com/Automattic/woocommerce-payments/assets/3147296/6947f200-6b99-4ab8-a4cd-3893e40df300">
* View a non-disputed transaction details screen.
* Ensure the Fees tooltip does not render.

-------------------


**Post merge**
